Odata-docs/client/grouping-and-aggregation.md

```diff
@@ -0,0 +1,485 @@
+    ```
+    /Sales?$apply=aggregate(Product/TaxRate with average as AverageProduct_TaxRate)
+    ```
+    _Result:_ 0.1
+
+6. Return sum sale product tax rate - where Product is a navigation property
+
+    _Expression:_
+    ```csharp
+    dataServiceContext.Sales.Sum(d1 => d1.Product.TaxRate)
+    ```
+    _Translation:_
+    ```
+    /Sales?$apply=aggregate(Product/TaxRate with sum as SumProduct_TaxRate)
+    ```
+    _Result:_ 0.8
+
+7. Return minimum sale product tax rate - where Product is a navigation property
+
+    _Expression:_
+    ```csharp
+    dataServiceContext.Sales.Min(d1 => d1.Product.TaxRate)
+    ```
+    _Translation:_
+    ```
+    /Sales?$apply=aggregate(Product/TaxRate with min as MinProduct_TaxRate)
+    ```
+    _Result:_ 0.06
+
+8. Return maximum sale product tax rate - where Product is a navigation property
+
+    _Expression:_
+    ```csharp
+    dataServiceContext.Sales.Max(d1 => d1.Product.TaxRate)
+    ```
+    _Translation:_
+    ```
+    /Sales?$apply=aggregate(Product/TaxRate with max as MaxProduct_TaxRate)
+    ```
+    _Result:_ 0.14
+
+In addition to the 4 aggregation methods described in the section above, we also support a custom [CountDistinct](/dotnet/api/microsoft.odata.client.dataserviceextensions.countdistinct) method that returns the total number of distinct values in the specified property. This custom method was introduced to eliminate the need for writing an expression of the following form to achieve the same: `dataServiceContext.Sales.Select(d1 => d1.Product).Distinct().Count()`.
+
+### CountDistinct Examples
+1. CountDistinct targeting structural property
+
+    _Expression:_
+    ```csharp
+    dataServiceContext.Sales.CountDistinct(d1 => d1.ProductId)
+    ```
+    _Translation:_
+    ```
+    /Sales?$apply=aggregate(ProductId with countdistinct as CountDistinctProductId)
+    ```
+    _Result:_ 3
+
+2. CountDistinct targeting navigation property
+
+    _Expression:_
+    ```csharp
+    dataServiceContext.Sales.CountDistinct(d1 => d1.Product.Color)
+    ```
+    _Translation:_
+    ```
+    /Sales?$apply=aggregate(Product/Color with countdistinct as CountDistinctProductColor)
+    ```
+    _Result:_ 2
+
+We can restrict the set of data to be aggregated by using the `Where` method. In this case however, the `Where` method is translated into a `filter` transformation, not a `$filter` query option.
+
+### Filter Transformation Example
+Return average USD sale amounts
+
+_Expression:_
+```csharp
+dataServiceContext.Sales.Where(d1 => d1.CurrencyCode == "USD").Average(d2 => d2.Amount)
+```
+_Translation:_
+```csharp
+/Sales?$apply=filter(CurrencyCode eq 'USD')/aggregate(Amount with average as AverageAmount)
+```
+_Result:_ 3.8
+
+## Grouping and Aggregation
+There are [8 overloads](/dotnet/api/system.linq.queryable.groupby) of the `GroupBy` method defined in the `System.Linq` library. Our starting point in implementing support for grouping and aggregation is the following workload:
```

Comment:
Hey, @gathogojr what does workload refer to here?

Reply:
It should be overload. Nice catch